rs/sns/governance/src/governance.rs

```diff
@@ -4748,6 +4822,114 @@ impl Governance {
             }
         }
     }
+    /// Checks if an automatic upgrade is needed and initiates it.
+    /// An automatic upgrade is needed if `target_version` is set to a future version on the upgrade path
+    async fn initiate_upgrade_if_sns_behind_target_version(&mut self) {
+        // Check that no upgrades are in progress
+        if self.check_no_upgrades_in_progress(None).is_err() {
```

Comment:
It seems an upgrade proposal could be adopted and be "in progress" while awaiting the root to upgrade the SNS. Is there a lock preventing this?

Reply:
`check_no_upgrades_in_progress` is exactly what detects this case and prevents us from initiating an upgrade if necessary